go-controller/pkg/ovn/master_test.go

```diff
@@ -1678,6 +1679,167 @@ var _ = ginkgo.Describe("Default network controller operations", func() {
+
+			return nil
+		}
+
+		err := app.Run([]string{
+			app.Name,
+			"-cluster-subnets=" + clusterCIDR,
+			"--init-gateways",
+			"--nodeport",
+		})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+
+	ginkgo.It("verify IPs in host-network-namespace address_set after remote zone node deletion", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := config.InitConfig(ctx, nil, nil)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			config.OVNKubernetesFeature.EnableInterconnect = true
+			config.Kubernetes.HostNetworkNamespace = "ovn-host-network"
+
+			// PortBinding is required to add interconect resources for
+			// remote zone nodes
+			pb := &sbdb.PortBinding{
+				LogicalPort: "tstor-node1",
+				Datapath:    "tstor-node1",
+				TunnelKey:   5,
+			}
+			dbSetup.SBData = append(
+				dbSetup.SBData,
+				pb,
+			)
+
+			// Set annotations required to create remote zone node
+			newNodeSubnet := "10.1.2.0/24"
+			transitswitchSubnet := "100.88.0.3/16"
+			testNode.Annotations["k8s.ovn.org/node-subnets"] = fmt.Sprintf("{\"default\":[\"%s\"]}", newNodeSubnet)
+			testNode.Annotations["k8s.ovn.org/network-ids"] = fmt.Sprintf("{\"default\":\"%d\"}", 0)
+			testNode.Annotations["k8s.ovn.org/node-chassis-id"] = "2"
+			testNode.Annotations["k8s.ovn.org/l3-gateway-config"] = `{"default":{"mac-address":"0a:58:01:02:03:04","ip-addresses":["1.2.3.4/24"],"ip-address":"1.2.3.4/24","next-hops":["1.2.3.1"],"next-hop":"1.2.3.1"}}`
+			testNode.Annotations["k8s.ovn.org/node-transit-switch-port-ifaddr"] = fmt.Sprintf("{\"ipv4\":\"%s\"}", transitswitchSubnet)
+			testNode.Annotations["k8s.ovn.org/node-mgmt-port-mac-address"] = "0a:58:0a:01:01:02"
+			testNode.Annotations["k8s.ovn.org/zone-name"] = "foo"
+			fakeOvn.startWithDBSetup(dbSetup,
+				&v1.NamespaceList{
+					Items: []v1.Namespace{*newNamespace(config.Kubernetes.HostNetworkNamespace)},
+				},
+
+				&v1.NodeList{
+					Items: []v1.Node{testNode},
+				},
+			)
+
+			// Required to make sure a remote zone node is being added
+			fakeOvn.controller.zone = "node1"
+
+			gomega.Expect(fakeOvn.controller.WatchNamespaces()).To(gomega.Succeed(), "Namespace should be created fine")
+			err = fakeOvn.controller.WatchNodes()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			// Extract expected IPs and check address_set
+			var ips []string
+			hostSubnets, err := util.ParseNodeHostSubnetAnnotation(&testNode, types.DefaultNetworkName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			mgmt_ips := util.GetNodeManagementIfAddr(hostSubnets[0])
+			ips = append(ips, mgmt_ips.IP.String())
+			lrpips, err := util.ParseNodeGatewayRouterLRPAddrs(&testNode)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			lrpip, _, _ := net.ParseCIDR(lrpips[0].String())
+			ips = append(ips, lrpip.String())
+			fakeOvn.asf.ExpectAddressSetWithIPs(config.Kubernetes.HostNetworkNamespace, ips)
+
+			// Delete the node and check whether the address_set is empty or not
```

Comment:
do you really need a new test for this? just rename previous test to add/delete whatever and then delete the node after the add and just check if IPs are getting removed? That saves up so much duplicate test set up code and test run time...